### PR TITLE
Auto associations: Fixes for hasMany and belongsTo.

### PR DESCRIPTION
**Pull request by Fridus**
- Fixes for hasMany and belongsTo.
- Add tests

### CODE REVIEW

File: tests/associations/has-one.test.js [hunk 1]
```diff
@@ -67,12 +67,26 @@ describe('Associations(HasOne)', function() {
...
+        }),
+        test.models.User.create({
+          username: 'MannekenPis',
+          email: 'manneken.pis@brussels.be'
         })
-      ]).spread(function(address, user) {
-        return user.setAddress(address);
+      ]).spread(function(address, address2, user, user2) {
+        return user.setAddress(address).then(function(){
+          user2.setAddress(address2).then(function(){
```
Comment (mbroadst) [inline tests/associations/has-one.test.js]: you should be able to just do:

```
return user2.setAddress(address2).then(function() {
```

and get rid of the ```done();```

**Comment by Fridus** [comment]
I started working on association of type `belongsToMany` in https://github.com/Fridus/epilogue/tree/auto-associations-belongstomany

File: tests/associations/has-one.test.js (hunk 1 above)
Comment (Fridus) [inline tests/associations/has-one.test.js]: I do that for fix the error on node v0.10: https://github.com/Fridus/epilogue/commit/50142a483f46dd14741f7e9fdd18dc34bd407067
https://travis-ci.org/dchester/epilogue/builds/64384659

This works on my desktop:
```
... .spread(function(address, address2, user, user2) {
        user.setAddress(address);
        return user2.setAddress(address2);
      });
```

Comment (mbroadst) [inline tests/associations/has-one.test.js]: oh sorry I didn't read this properly the first time, you should just:

```
return Promise.all([
  user.setAddress(address),
  user2.setAddress(address2)
]);
```

**Comment by Fridus** [comment]
I have fixed like that:
```
.spread(function(address, address2, user, user2) {
  return user.setAddress(address).then(function() {
    return user2.setAddress(address2);
});
```

**Pull request merged**